app/controllers/incomeSources/add/ForeignPropertyCheckDetailsController.scala

```diff
@@ -0,0 +1,203 @@
+                                                      val appConfig: FrontendAppConfig,
+                                                      implicit val itvcErrorHandler: ItvcErrorHandler,
+                                                      implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler) extends ClientConfirmedController
+  with FeatureSwitching {
+
+  lazy val foreignPropertyAccountingMethodUrl: String = controllers.incomeSources.add.routes.ForeignPropertyAccountingMethodController.show().url
+  lazy val agentForeignPropertyAccountingMethodUrl: String = controllers.incomeSources.add.routes.ForeignPropertyAccountingMethodController.showAgent().url
+  lazy val backUrlIndividual: String = foreignPropertyAccountingMethodUrl
+  lazy val backUrlAgent: String = agentForeignPropertyAccountingMethodUrl
+
+  def handleRequest(sources: IncomeSourceDetailsModel, isAgent: Boolean)
+                   (implicit user: MtdItUser[_], hc: HeaderCarrier): Future[Result] = {
+
+    val backUrl: String = if (isAgent) backUrlAgent else backUrlIndividual
+    val postAction: Call = if (isAgent) controllers.incomeSources.add.routes.ForeignPropertyCheckDetailsController.submitAgent() else
+      controllers.incomeSources.add.routes.ForeignPropertyCheckDetailsController.submit()
+
+    if (isDisabled(IncomeSources)) {
+      if (isAgent) Future.successful(Redirect(controllers.routes.HomeController.showAgent)) else Future.successful(Redirect(controllers.routes.HomeController.show()))
+    } else {
+      Future {
+        getDetails(user) match {
+          case Right(viewModel) =>
+            Ok(checkForeignPropertyDetails(
+              viewModel,
+              postAction = postAction,
+              isAgent,
+              backUrl = backUrl
+            ))
+          case Left(ex) =>
+            if (isAgent) {
+              Logger("application").error(
+                s"[Agent][ForeignPropertyCheckDetailsController][handleRequest] - Error: ${ex.getMessage}")
+              itvcErrorHandlerAgent.showInternalServerError()
+            } else {
+              Logger("application").error(
+                s"[ForeignPropertyCheckDetailsController][handleRequest] - Error: ${ex.getMessage}")
+              itvcErrorHandler.showInternalServerError()
+            }
+        }
+      }.recover{
+        case ex =>
+          if (isAgent) {
+            Logger("application").error(
+              s"[Agent][ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
+            itvcErrorHandlerAgent.showInternalServerError()
+          } else {
+            Logger("application").error(
+              s"[ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
+            itvcErrorHandler.showInternalServerError()
+          }
+      }
+    }
+  }
+
+  def getDetails(implicit user: MtdItUser[_]): Either[Throwable, CheckForeignPropertyViewModel] = {
```

Comment:
as per agreement we have recently in tech-channel*,
result of this function should be Future[Either[Throwable, CheckForeignPropertyViewModel]] - as this simplify integrating reading these data from Mongo and this is a new code